PCbuild/get_external.py

```diff
@@ -9,17 +9,36 @@
 from urllib.request import urlretrieve
 
 
+def retrieve_with_retries(download_location, output_path, reporthook,
+                          max_retries=7):
+    """Download a file with exponential backoff retry and save to disk."""
+    for attempt in range(max_retries):
```

Comment:
Backported off-by-one fix:
```suggestion
    for attempt in range(max_retries + 1):
```